pkg/provider/bitbucketdatacenter/bitbucketdatacenter.go

```diff
@@ -42,6 +44,48 @@ type Provider struct {
 	apiURL                    string
 	provenance                string
 	projectKey                string
+	repo                      *v1alpha1.Repository
+	triggerEvent              string
+}
+
+func (v Provider) Client() *bbv1.APIClient {
```

Comment:
why this is not a pointer receiver as the other methods?

```
   The methods `Client()` (and `ScmClient()` for Bitbucket DC)
    in these files have *value receivers* (e.g., `func (v Provider) Client()
    ...`). However, the `recordAPIUsageMetrics` method called within them
    attempts to lazily initialize the `v.metrics` field (`if v.metrics == nil {
    ...; v.metrics = m }`). Because the receiver is a value (a copy), any
    modification to `v.metrics` inside `recordAPIUsageMetrics` will *not*
    persist on the original `Provider` struct instance after the `Client()`
    method returns.
```